app/controllers/authController.js

```diff
@@ -75,6 +75,14 @@ class AuthController {
     const resMessage = await AuthService.resetPassword(newPassword, token);
     return ResponseHandler.success(res, resMessage);
   }
+
+  // send activate email to new user
+  async sendActivateEmail(req, res) {
+    const userId = req.body.user_id || req.headers['user_id']; // user_id is the _id in user model
```

Comment:
1. A basic empty check should be employed to ensure the userId is not empty and comply with the format required. This can be either implemented here or a better way is to add a validator in the route.
2. Why the userId can be retrieved from two sources, this would cause confusion to the front-end developer. Simple and accurate is way more important.
3. After skim through all the relative code, I can't understand the purpose of user IP?
4. One more question that you might need to think? How to avoid malicious active email that triggered not by the user email holder?

Reply:
Here I just followed the old version of this project. I think I can do it in a validator which will be better. 
UserIP is used to get a token, you can see the line 120 of the file authService.js.